servers/rendering/rasterizer_rd/shaders/scene_high_end.glsl

```diff
@@ -786,837 +786,712 @@ float sample_directional_soft_shadow(texture2D shadow, vec3 pssm_coord, vec2 tex
-			splane = (lights.data[idx].shadow_matrix * vec4(vertex - normalize(normal_interp) * lights.data[idx].transmittance_bias, 1.0));
-
-			shadow_len = length(splane.xyz);
-			splane = normalize(splane.xyz);
-
-			if (splane.z >= 0.0) {
-				splane.z += 1.0;
+	return color;
+}
 
-			} else {
-				splane.z = 1.0 - splane.z;
-			}
+vec4 voxel_cone_trace_45_degrees(texture3D probe, vec3 cell_size, vec3 pos, vec3 direction, float tan_half_angle, float max_distance, float p_bias) {
+	float dist = p_bias;
+	vec4 color = vec4(0.0);
+	float radius = max(0.5, tan_half_angle * dist);
+	float lod_level = log2(radius * 2.0);
 
-			splane.xy /= splane.z;
-			splane.xy = splane.xy * 0.5 + 0.5;
-			splane.z = shadow_len * lights.data[idx].inv_radius;
-			splane.xy = clamp_rect.xy + splane.xy * clamp_rect.zw;
-			splane.w = 1.0; //needed? i think it should be 1 already
+	while (dist < max_distance && color.a < 0.95) {
+		vec3 uvw_pos = (pos + dist * direction) * cell_size;
 
-			float shadow_z = textureLod(sampler2D(shadow_atlas, material_samplers[SAMPLER_LINEAR_CLAMP]), splane.xy, 0.0).r;
-			transmittance_z = (splane.z - shadow_z) / lights.data[idx].inv_radius;
+		//check if outside, then break
+		if (any(greaterThan(abs(uvw_pos - 0.5), vec3(0.5f + radius * cell_size)))) {
+			break;
 		}
-#endif
+		vec4 scolor = textureLod(sampler3D(probe, material_samplers[SAMPLER_LINEAR_WITH_MIPMAPS_CLAMP]), uvw_pos, lod_level);
+		lod_level += 1.0;
 
-		vec3 no_shadow = vec3(1.0);
+		float a = (1.0 - color.a);
+		scolor *= a;
+		color += scolor;
+		dist += radius;
+		radius = max(0.5, tan_half_angle * dist);
+	}
 
-		if (lights.data[idx].projector_rect != vec4(0.0)) {
-			vec3 local_v = (lights.data[idx].shadow_matrix * vec4(vertex, 1.0)).xyz;
-			local_v = normalize(local_v);
+	return color;
+}
 
-			vec4 atlas_rect = lights.data[idx].projector_rect;
+void gi_probe_compute(uint index, vec3 position, vec3 normal, vec3 ref_vec, mat3 normal_xform, float roughness, vec3 ambient, vec3 environment, inout vec4 out_spec, inout vec4 out_diff) {
+	position = (gi_probes.data[index].xform * vec4(position, 1.0)).xyz;
+	ref_vec = normalize((gi_probes.data[index].xform * vec4(ref_vec, 0.0)).xyz);
+	normal = normalize((gi_probes.data[index].xform * vec4(normal, 0.0)).xyz);
 
-			if (local_v.z >= 0.0) {
-				local_v.z += 1.0;
-				atlas_rect.y += atlas_rect.w;
+	position += normal * gi_probes.data[index].normal_bias;
 
-			} else {
-				local_v.z = 1.0 - local_v.z;
-			}
+	//this causes corrupted pixels, i have no idea why..
+	if (any(bvec2(any(lessThan(position, vec3(0.0))), any(greaterThan(position, gi_probes.data[index].bounds))))) {
+		return;
+	}
 
-			local_v.xy /= local_v.z;
-			local_v.xy = local_v.xy * 0.5 + 0.5;
-			vec2 proj_uv = local_v.xy * atlas_rect.zw;
+	vec3 blendv = abs(position / gi_probes.data[index].bounds * 2.0 - 1.0);
+	float blend = clamp(1.0 - max(blendv.x, max(blendv.y, blendv.z)), 0.0, 1.0);
+	//float blend=1.0;
```

Comment:
Why is this code commented?

Reply:
I did not change this code. I checked and this comment was also present in Godot 3.2 scene.glsl, so it was probably just forgotten. I will consider if I should do code cleanup in this commit in a future update.